src/include/serial_inline.h

```diff
@@ -279,7 +279,12 @@ __wt_update_serial(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_PAGE *page
 
         page->modify->obsolete_check_txn = WT_TXN_NONE;
     }
-
+    /* 
+     * We don't want to remove obsolete updates in the history store, 
+     * since another reader might be reading these updates.
```

Comment:
We can check this in front of the previous if section.

Reply:
done